### DISCUSSION TITLE
Change the visual of osu!catch editor's Distance Snap

### DISCUSSION THREAD

**Discussion by LonaGit** [ppy/osu#16448] [Ideas] [answered]
As of now, there is not really a good way for people mapping in osu!catch to visualize the timeline on the playfield on higher snap divisors. And when turning on Distance Snap, you get these grey lines, which also dont really indicate the timeline pretty well.
![unknown](https://user-images.githubusercontent.com/97619525/149507719-b12fd34f-161e-4489-a8a9-823618210ec2.png)

I think when using Distance Snap at least, a better visualization on the playfied for where on the timeline a note will be placed (much like standar's editor, where it circles out with the snap divisor colors), would be a helpfull thing to have. This will acurately show the timeline with the distance snap, and is super helpful, at least for high snap divisors (1/6 and up), as at the moment, it requires constant observation of the timeline itself when placing notes. The middle line should stay, of course, so we can still snap things right above a fruit without problems, but i think the diagonal lines should be changed to simple straight lines, with the color and height of their place on the timeline. Picture below for vizualisation of what i mean.
![unknown (1)](https://user-images.githubusercontent.com/97619525/149507939-defbd96a-2d9a-4eab-a2c6-09f4633d8214.png)
 
This should hopefuly also make it easier to deal with distance snap adjustments, which hopefully will come (https://github.com/ppy/osu/issues/16445)

**Comment by peppy** [comment]
I think this is a pretty good approach, potentially as smaller ticks on the diagonal line (perpendicular to the line to make it look nicer?) or maybe dots instead.

**Reply by LonaGit** [reply]
> I think this is a pretty good approach, potentially as smaller ticks on the diagonal line (perpendicular to the line to make it look nicer?) or maybe dots instead.

yeah that could work. dots might be a little hard to see depending on their size. but smaller ticks on the diagonal line i think would be the better one.

**Comment by snalgae** [comment]
What if there were horizontal lines across the field that would mark the beat of the music that you can align objects to?

**Reply by Xinnoh** [reply, marked as answer]
ah missed this thread
yes that would work
https://github.com/ppy/osu/discussions/23462

**Reply by peppy** [reply]
I'd still like to revisit the visuals of the distance snap grid though, I'm not a huge fan of the current display.
